src/org/jitsi/impl/neomedia/transform/rewriting/SsrcGroupRewriter.java

```diff
@@ -414,11 +415,25 @@ private void maybeSwitchActiveRewriter(final RawPacket pkt)
      */
     boolean isKeyFrame(RawPacket pkt)
     {
-        int sourceSSRC = pkt.getSSRC();
-        Byte redPT = ssrcRewritingEngine.ssrc2red.get(sourceSSRC);
-        byte vp8PT = 0x64;
-
-        return Utils.isKeyFrame(pkt, redPT, vp8PT);
+        byte pt = pkt.getPayloadType();
+        Byte redPT = null, codecPT = null;
+        String codecType = null;
+        for (Map.Entry<Byte, MediaFormat> entry :
```

Comment:
We have this logic twice - the dynamic payload types loop that detects a keyframe -. Once here and once in the RTCPFeedbackMessageSender class. Maybe we could move it inside the MediaStreamImpl?